.github/workflows/test-http.yml

```diff
@@ -0,0 +1,35 @@
+name: "Black Box Testing"
+
+# run this on a pull request for testing
+on:
+  pull_request:
+
+# use a single job at first
+jobs:
+  run-functional-tests:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v2
+    - name: Build and Run Services and Mongo Containers
+      run: |
+        cp docker/.docker-env.example docker/.docker-env
+        docker-compose --file docker/docker-compose.yml build
+        docker-compose --file docker/docker-compose.yml up &
+    - name: Build Test Container
+      run: |
+        cd test-http/
+        cp docker/.docker-env.example docker/.docker-env
+        docker-compose --file docker/docker-compose.yml build
+    - name: Run Test Container
+      run: |
+        cd test-http/
+        docker-compose --file docker/docker-compose.yml up &
+    - name: Sleep
+      run: sleep 30
+    - name: Load Data into MongoDb
+      run: docker exec cveawg npm run populate:dev
+    - name: Run Black Box Tests
+      run: |
+        cd test-http/
+        docker exec demon pytest src/
```

Comment:
This step will _always_ pass, because the `docker exec` command succeeds even if the tests fail. It needs to be updated to parse the Pytest output, look for the "passed, errored, failed" results line, and fail if an error or a failure occurs in the tests.

Reply:
It fails for me. I added the extra checks either way, but since the `docker exec` command fails if python tests fail the extra checks will be skipped.

<img width="1680" alt="Screen Shot 2021-06-22 at 5 17 22 PM" src="https://user-images.githubusercontent.com/5251788/123002163-98098480-d37f-11eb-8370-8f98aee332c6.png">